ui/client/datasets/DataTransformation/AdjustGeoResolution.js

```diff
@@ -70,47 +100,93 @@ export default withStyles((theme) => ({
     return args;
   }, [selectedResolution, oldResolution]);
 
+  const formatNewResolution = () => {
+    if (!selectedResolution || selectedResolution === '') return '';
+    if (nonUniform) {
+      if (validNumber(selectedResolution)) {
+        return `${selectedResolution} km`;
+      }
+
+      return '';
+    }
+    return `${selectedResolution.toFixed(2)} km`;
+  };
+
+  const disableButton = () => {
```

Comment:
I don't think I understand why this is a function and not a boolean. Would be helpful if you leave a note on the code, or maybe we can talk through it here.

eg what I have in my mind:

```
const areTransformButtonsDisabled = (nonUniform && !validNumber(selectedResolution)) || !selectedResolution;
```